microprofile/WFLY-14987_reactive_messaging-ssl-context-kafka-connector.adoc

```diff
@@ -0,0 +1,278 @@
+= The MP Reactive Messaging Kafka connector should be able to use an SSLContext configured in the Elytron subsystem
+:author:            Kabir Khan
+:email:             kkhan@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+The SmallRye Kafka Connector for MicroProfile Reactive Messaging uses the Apache Kafka client under the hood.
+
+To connect to a secured instance of Kafka which doesn't have a CA signed certificate you need to specify MicroProfile Config properties to configure the truststore, either deployment-wide in `microprofile-config.properties`, e.g.
+```
+# mp.messaging.connector.smallrye-kafka.ssl.truststore.location=/Users/kabir/temp/kafka_2.13-2.8.0/config/sasl-ssl/client.truststore.p12
+# mp.messaging.connector.smallrye-kafka.ssl.truststore.password=clientts
+```
+or override per `@Incoming` (or `@Outgoing`, not shown - it follows the same pattern) annotated method in `microprofile-config.properties`:
+```
+mp.messaging.outgoing.to-kafka.ssl.truststore.location=/Users/kabir/temp/kafka_2.13-2.8.0/config/sasl-ssl/client.truststore.p12
+mp.messaging.outgoing.to-kafka.ssl.truststore.password=test
+```
+Under the hood the prefixes are stripped off by SmallRye Reactive Messaging, and the keys that are passed in to the Kafka client are simply `ssl.truststore.location` and `ssl.truststore.password`.
+
+These properties are used by the Apache Kafka client as input for creating a `javax.net.ssl.SSLContext`, which in turn is used to create a `javax.net.ssl.SSLEngine` for the secure connection.
+
+This RFE proposes that in addition to the above approach to configure the `SSLContext`, we also provide the ability to set up an `SSLContext` via the `/subsystem=elytron/client-ssl-context=*` management resources in WildFly and to have a custom property to make the Kafka client connection use such an SSLContext.
```

Comment:
Do I understand correctly that this is happening in WildFly integration and not in Smallrye MP Reactive messaging implementation? In other words Smallrye implementation does not depend on Elytron. And in other words this feature is cant be used somewhere else, e.g. Quarkus.

Reply:
Yes, this is all in WildFly